### DISCUSSION TITLE
User를 삭제하면 어떤 상태 코드를 반환하는 것이 좋을까?

### DISCUSSION THREAD

**Discussion by beatmeJY** [dailyge/dailyge-server#81] [Information]
## 🔗 연관 PR
- [[DLG-13] User 삭제 API 기능을 재 수정한다.](https://github.com/dailyge/dailyge-server/pull/7)

<br/><br/>

## ❓ 작성 이유
API를 통해 자원을 삭제할 경우 어떠한 응답을 반환하여 주는 것이 좋을지에 대해 의견을 나누어 최종 결과를 정리하고 팀 내에 공유하기 위해 작성하였습니다.

<br/><br/>

### 탈퇴 시 응답코드로 `200`의 의견
 - [해당 글](https://hyeonproject.medium.com/http-status-code-204%EC%9D%98-%EC%9E%AC%EB%B0%8C%EB%8A%94-%EC%A0%90-dbb7e8ef26da)을 보면 204 No Content는 Body에 아무런 값을 보내 줄 것이 없다는 의미이기에 사용하지 않았다.
 - 회원 탈퇴를 할 경우 회원 정보 페이지에서 벗어나야 하기 때문에 어울리지 않는다고 생각하였다.
     -  [MDM 204](https://developer.mozilla.org/ko/docs/Web/HTTP/Status/204): 상태 응답 코드는 요청이 성공했으나 클라이언트가 현재 페이지에서 벗어나지 않아도 된다는 것을 나타냅니다."

<br/>

###  탈퇴 시 응답코드로 `204`의 의견
 -  200도 틀린 것은 아니지만, 탈퇴 시에 굳이 추가 메시지를 내려 줄 필요가 있을까 생각하였다.
     - [MDM DELETE 200](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE#responses): 작업이 실행되었고 응답 메시지에 상태를 설명하는 표현이 포함된 경우 200(확인) 상태 코드입니다.
    - [MDM DELETE 204](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE#responses): 작업이 실행되었지만 추가 정보를 제공할 필요가 없는 경우 204(내용 없음) 상태 코드입니다.
- 탈퇴의 경우 사용자 탈퇴 후, 로그아웃 API를 호출해서 성공하면 200 OK를 주고 그 때 다른 페이지로 이동 시키는 게 각 API가 하는 역할이 명확하다.
    - 이때, 단순히 로그아웃 API를 호출하면, 삭제된 사용자일 때 404 NOT_FOUND 가 발생할 수 있으니, 
    - 탈퇴했을 때 처리하는 별도의 API를 만들거나, 별도의 과정은 조금 필요하다.

<br/>
<br/>

### 최종 결정은 `204`.
> 회원 탈퇴 시 추가적인 메시지를 내려줄 것이 없다 판단하여 204를 사용하기로 결정하였습니다.

**Comment by devjun10** [comment]
정리 너무 좋네요. 👍
